File: CODEFORCES/CONTEST/Codeforces Round 898 (Div. 4)/F.py
import sys;input=sys.stdin.readline;R=lambda:list(map(int,input().split()))
t = int(input())
for _ in range(t):
    n,k=R()
    a=R()
    h=R()
    if min(a)>k: print(0); continue
    res=1
    mask=0
    fr=a[0]
    i=1
    j=0
    while i<n:
        if h[i-1]%h[i]==0:
            fr+=a[i]
            if fr>k:
                j=mask
                while fr>k:
                    fr-=a[j]
                    j+=1
                mask=j
            res=max(res,i-mask+1)
        else:
            fr=a[i]
            mask=i
       # print(fr,mask,i)
        i+=1
    print(res)
        

File: CONTEST/CODEFORCES/Codeforces Round 905 (Div. 3)/F.cpp
#include<bits/stdc++.h>
using namespace std;

#define int long long
#define ll long long
#pragma GCC optimize("O3,expensive-optimizations,Ofast,unroll-loops")
#pragma GCC target("sse,sse2,sse3,ssse3,sse4,popcnt,abm,mmx,avx,tune=native")
#define reset(a,n) for( int i = 0; i <= n; i++) a[i] = 0
const int N   = 1e6 + 10;
const int mod = 479001599;
const int MAX = INFINITY;

clock_t startTime;
double getCurrentTime() {
	return (double)(clock() - startTime) / CLOCKS_PER_SEC;
}

void solve(){ 
    int n; 
    cin >> n;
    int a[n + 1];
    map<int, int> cnt;
    for (int i = 1; i <= n; i++ ){
        cin >> a[i];
        cnt[a[i]]++;
    }
    int res = 0, cur = cnt.size();
    for (auto x: cnt) res += (x.second == 1);
    map<int, bool> check;
    for (int i = 1; i <= n; i++){
        cnt[a[i]]--;
        if (!cnt[a[i]]) cur--;
        if (check[a[i]]) continue;
        check[a[i]] = 1;
        //cout << res << ' ' << cur << endl;
        res += cur;
    }
    cout << res;
}
 
signed main(){
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr); cout.tie(nullptr);
    int local = 0;
	if(fopen("input.txt", "r")){
        freopen("input.txt", "r", stdin); 
        freopen( "output.txt", "w", stdout);
        local = 1;
    }
    int tc = 1; 
    cin >> tc;
    while (tc--) {
        solve();
        cout << endl;
    }
    if (local) cout << getCurrentTime();
    return 0;
}